package/src/error.rs

```diff
@@ -133,7 +165,49 @@ impl std::fmt::Display for Error {
             Error::LockFileDeserialization { path, error } => {
                 write!(f, "lock file {} is invalid: {error}", path.display())
             }
-            Error::IndexDep => write!(f, "index dependencies are not yet implemented"),
+            Error::UnknownIndexPackage { id } => write!(f, "package {id} not found in the index"),
+            Error::UnknownIndexPackageVersion {
+                id,
+                requested,
+                available,
+            } => {
+                let available: Vec<_> = available.iter().map(|x| x.to_string()).collect();
+                write!(
+                    f,
+                    "package {id}@{requested} not found in the index. Available versions: {}",
+                    available.join(", ")
+                )
+            }
+            Error::InvalidIndexDep { id, dep } => match dep.as_ref() {
+                UnversionedDependency::Git(g) => write!(
+                    f,
+                    "package {id} depends on git package {}, so it cannot be put in the index",
+                    g.url
+                ),
+                UnversionedDependency::Path(path) => write!(
+                    f,
+                    "package {id} depends on path package {}, so it cannot be put in the index",
+                    path.display()
+                ),
+            },
+            Error::DuplicateIndexPackageVersion { id, version } => {
+                write!(f, "package {id}@{version} is already present in the index")
+            }
+            Error::PackageIndexSerialization { error, pkg } => {
+                write!(f, "error serializing package. Failed package {pkg:?}, caused by {error}\n{INTERNAL_ERROR_MSG}")
```

Comment:
There is some duplication in the error message. Maybe it's a mix of two different versions?